server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -373,15 +369,9 @@ private HitContext prepareNestedHitContext(SearchContext context,
             for (SearchHit.NestedIdentity nested = nestedIdentity; nested != null; nested = nested.getChild()) {
                 String nestedPath = nested.getField().string();
                 current.put(nestedPath, new HashMap<>());
-                List<?> nestedParsedSource = XContentMapValues.extractNestedValue(nestedPath, rootSourceAsMap);
-                if ((nestedParsedSource.get(0) instanceof Map) == false && hasNonNestedParent.test(nestedPath)) {
-                    // When one of the parent objects are not nested then XContentMapValues.extractValue(...) extracts the values
-                    // from two or more layers resulting in a list of list being returned. This is because nestedPath
-                    // encapsulates two or more object layers in the _source.
-                    //
-                    // This is why only the first element of nestedParsedSource needs to be checked.
-                    throw new IllegalArgumentException("Cannot execute inner hits. One or more parent object fields of nested field [" +
```

Comment:
Was the only reason we used to error here the list-of-list structure that was a result of the way we parse the source map? I'd like to double check to understand if flattening now doesn't create problems elsewhere. Still trying to understand the reason for this error in the past better, maybe you can explain this in a bit more detail?

Reply:
Yes, we tried to parse things using ObjectMapper lookups before, which would fail if there was an object mapper somewhere in the hierarchy that wasn't a nested mapper.  This changes the logic of extraction entirely to be based on the path and then looking for maps, so we don't have this restriction any longer.